src/content/developers/docs/oracles/index.md

```diff
@@ -8,6 +8,211 @@ incomplete: true
 
 Oracles are a data feed that connect Ethereum to off-chain, real-world information, so you can query data in your smart contracts. For example, prediction market dapps use oracles to settle payments based on events. A prediction market may ask you to bet your ETH on the next president of the United States. They'll use an oracle to confirm the outcome and pay out to the winners.
 
+## Prerequisites {#prerequisites}
+
+Make sure you're familiar with [nodes](/developers/docs/nodes-and-clients/), [consensus mechanisms](/developers/docs/consensus-mechanisms/), and [smart contract anatomy](/developer/docs/smart-contracts/anatomy/), specifically events.
+
+## What is an oracle {#what-is-an-oracle}
+
+Oracles act as a bridge between the blockchain and the real world. They allow you to query APIs to get information into your smart contracts. This could be anything from price information to weather reports.
+
+## Why are they needed? {#why-are-they-needed}
+
+With a blockchain like Ethereum you need every node in the network to be able to replay every transaction and end up with the same result, guaranteed. APIs introduce potentially variable data. If you were sending someone an amount of ETH based on an agreed \$USD value using a price API, the query would return a different result from one day to the next. Not to mention, the API could be hacked or deprecated. If this happens, the nodes in the network wouldn't be able to agree on Ethereum's current state, effectively breaking [consensus](/developers/docs/consensus-mechanisms/).
+
+Oracles solve this problem by posting the data on the blockchain. So any node replaying the transaction will use the same immutable data that's posted for all to see. To do this, an oracle is made up of a smart contract and some off-chain components that can query APIs.
+
```

Comment:
I think it'd be great (perhaps in a v2) to have a "use cases" section here. A few that come to mind:
- Price feeds (e.g. https://medium.com/compound-finance/announcing-compound-open-oracle-development-cff36f06aad3 https://chain.link/solutions/defi)
- Randomness generation (e.g. https://blog.chain.link/verifiable-random-functions-vrf-random-number-generation-rng-feature/ https://chain.link/solutions/chainlink-vrf)
- Proof of reserve (e.g. https://cointelegraph.com/news/1b-in-wrapped-bitcoin-now-being-audited-using-chainlink-s-proof-of-reserve)